### PR TITLE
Changes to deactivate/reactivate products

### PR DESCRIPTION
**Pull request by rleed**
Draft changes to to support deactivation/reactivation of products per comment on issue #66. First time I've tinkered with this code, and I'm not sure how to add the toggle button to the UI, but maybe these suggested changes help somebody get a head start with this feature?

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -257,20 +258,24 @@ def to_nostr_event(self, pubkey: str) -> NostrEvent:
...
+                created_at=round(time.time()),
+                kind=30018,
+                tags=[["d", self.id]] + categories,
+                content=json.dumps(content, separators=(",", ":"), ensure_ascii=False),
+            )
+            event.id = event.event_id
+            
+            return event
+        else
+            return to_nostr_delete_event(self, str)
```
Comment (motorina0) [inline models.py:278]: - no need to pass `self` as a parameter

**Comment by motorina0** [comment]
Looks good so far. Thank you for opening the PR!
Do you plan to add the UI toggle as well?

**Comment by rleed** [comment]
> Do you plan to add the UI toggle as well?

Would like to try, but I'm not up to speed on the code base yet...could you point me to the right file? My idea would be to add a third button to go with the trashcan/delete and pencil/edit buttons, which would toggle the active state.

image

**Comment by motorina0** [comment]
- that is good place on the UI
- fIle: [static/components/stall-details/stall-details.html](https://github.com/lnbits/nostrmarket/blob/main/static/components/stall-details/stall-details.html#L97-L102)
- for toggle see the Activate/Deactivate component
- [updateProduct()](https://github.com/lnbits/nostrmarket/blob/main/static/components/stall-details/stall-details.js#L218C8-L218C8) method is already there, it just needs to be called with the right value

**Comment by motorina0** [comment]
- Is the PR ready for review? I see its not in `Draft` anymore.
- Please also take care to not fulfill orders that have disabled products.

**Comment by rleed** [comment]
> * Is the PR ready for review? I see its not in `Draft` anymore.

I thought it was, but I've noticed disabled products re-appear (even though they stay disabled on the merchant side). So maybe I need to track that down first.

> * Please also take care to not fulfill orders that have disabled products.

Good idea, although I guess that should not be a normal occurrence.

**Comment by rleed** [comment]
Question... with this change, I get an error (below) on false values but not on true values:

image

Any idea why?

image

It's probably something dumb, but I'm having trouble tracking it down as I'm not seeing where the validation is being done in the code anywhere.

**Comment by motorina0** [comment]
> Any idea why?

- the error message suggests that the `tags` field should be present (even if it is an empty list)

**Comment by motorina0** [comment]
@rleed is this PR ready? will you continue on it or do you want me to take over (fine with me)?

**Comment by rleed** [comment]
> @rleed is this PR ready? will you continue on it or do you want me to take over (fine with me)?

If you would be so kind as to take it over, that would be great. (I'm still missing a lot of understanding of the nostr library and vue.)

**Comment by rleed** [comment]
> @rleed is this PR ready?

It works as committed, but editing a product results in the active switch getting set. I was trying to fix that when I ran into the nostr error above.

**Comment by rleed** [comment]
I see this was closed without merge or comment. Was this feature implemented elsewhere?

**Pull request merged**